src/savegame_json.cpp

```diff
@@ -715,9 +704,8 @@ void player::deserialize(JsonIn &jsin)
         data.read( "grab_type", grab_typestr);
     }
 
-    if ( obj_type_id.find(grab_typestr) != obj_type_id.end() ) {
-        grab_type = (object_type)obj_type_id[grab_typestr];
-    }
+    const auto iter = std::find( obj_type_name.begin(), obj_type_name.end(), grab_typestr );
+    grab_type = static_cast<object_type>( std::distance( obj_type_name.begin(), iter ) );
```

Comment:
`iter` can point at `obj_type_name.end()` if no `grab_typestr` was found. In which case `grab_type` gets assigned to `NUM_OBJECTS`, but AFAIK our code doesn't expect that.